geomesa-features/geomesa-feature-kryo/src/main/scala/org/locationtech/geomesa/features/kryo/impl/KryoFeatureSerialization.scala

```diff
@@ -47,155 +49,191 @@ trait KryoFeatureSerialization extends SimpleFeatureSerializer {
+  }
+
+  case object KryoIntWriter extends KryoAttributeWriter {
+    override def apply(output: Output, value: AnyRef): Unit = output.writeInt(value.asInstanceOf[Int])
+  }
+
+  case object KryoLongWriter extends KryoAttributeWriter {
+    override def apply(output: Output, value: AnyRef): Unit = output.writeLong(value.asInstanceOf[Long])
+  }
+
+  case object KryoFloatWriter extends KryoAttributeWriter {
+    override def apply(output: Output, value: AnyRef): Unit = output.writeFloat(value.asInstanceOf[Float])
+  }
+
+  case object KryoDoubleWriter extends KryoAttributeWriter {
+    override def apply(output: Output, value: AnyRef): Unit = output.writeDouble(value.asInstanceOf[Double])
+  }
+
+  case object KryoBooleanWriter extends KryoAttributeWriter {
+    override def apply(output: Output, value: AnyRef): Unit = output.writeBoolean(value.asInstanceOf[Boolean])
+  }
+
+  case object KryoDateWriter extends KryoAttributeWriter {
+    override def apply(output: Output, value: AnyRef): Unit = output.writeLong(value.asInstanceOf[Date].getTime)
+  }
+
+  case object KryoUuidWriter extends KryoAttributeWriter {
+    override def apply(output: Output, value: AnyRef): Unit = {
+      val uuid = value.asInstanceOf[UUID]
+      output.writeLong(uuid.getMostSignificantBits)
+      output.writeLong(uuid.getLeastSignificantBits)
+    }
+  }
+
+  case object KryoJsonWriter extends KryoAttributeWriter {
+    override def apply(output: Output, value: AnyRef): Unit =
+      KryoJsonSerialization.serialize(output, value.asInstanceOf[String])
+  }
+
+  case object KryoBytesWriter extends KryoAttributeWriter {
+    override def apply(output: Output, value: AnyRef): Unit = {
+      val array = value.asInstanceOf[Array[Byte]]
+      output.writeInt(array.length, true)
+      output.writeBytes(array)
+    }
+  }
+
+  case object KryoGeometryWkbWriter extends KryoAttributeWriter {
+    override def apply(output: Output, value: AnyRef): Unit =
+      KryoGeometrySerialization.serializeWkb(output, value.asInstanceOf[Geometry])
+  }
+
+  case class KryoGeometryTwkbWriter(precision: GeometryPrecision.TwkbPrecision) extends KryoAttributeWriter {
+    override def apply(output: Output, value: AnyRef): Unit =
+      KryoGeometrySerialization.serialize(output, value.asInstanceOf[Geometry], precision)
+  }
+
+  case class KryoListWriter(elements: KryoAttributeWriter) extends KryoAttributeWriter {
+    // TODO handle null elements?
+    override def apply(output: Output, value: AnyRef): Unit = {
+      val list = value.asInstanceOf[java.util.List[AnyRef]]
+      output.writeInt(list.size(), true)
+      val iter = list.iterator()
+      while (iter.hasNext) {
+        elements(output, iter.next())
+      }
+    }
+  }
+
+  case class KryoMapWriter(keys: KryoAttributeWriter, values: KryoAttributeWriter) extends KryoAttributeWriter {
+    // TODO handle null keys/values?
```

Comment:
Do we have unit tests which cover this?

Reply:
no, same as above, do you think we should support it?